src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -447,7 +468,9 @@ private static void PopulatePropertyBag(SerializationPropertyBag propertyBag, in
         {
             foreach (var (property, serializationMapping) in propertyBag.Properties.ToArray())
             {
-                var schemaProperty = property.SchemaProperty!; // we ensure this is not null when we build the array
+                if (property.SchemaProperty == null)
+                    continue; // skip properties not from specification
+                Property schemaProperty = property.SchemaProperty;
```

Comment:
we should not skip the property when it is not from the specification.
The attribute `CodeGenSerialization` is able to assign an array of serialization names which mean the full serialization path

if you check the code below, the schemaProperty variable is barely used, we only use it as fallback value when the serializationMapping is null. When the property comes from the customization only, this mapping cannot be null, otherwise we will not have it here.

Reply:
fixed.